invoke Base Test Suite - java.lang.AssertionError: Unexpected status code. expected <br>[200] but found <br>[404]

I am using the WMS : http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities

The rest of the test passes correctly, and I am not sure why the this test is failing? Do I need to return application/xml as the response format and not text/xml? Not sure why this test is failing. I am testing on the beta validation site.

@dstenger @chuck1987 


I have tested this ets locally but I am getting the following error:

```
Test Name : tns:Main type Mandatory
net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public java.lang.Object com.occamlab.te.TECore.callFunction(net.sf.saxon.expr.XPathContext,java.lang.String,java.lang.String,net.sf.saxon.om.NodeInfo) throws java.lang.Exception}: Exception in extension function net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.RuntimeException: Failed to parse resource located at http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:192)
	at com.occamlab.te.TECore.executeTemplate(TECore.java:600)
	at com.occamlab.te.TECore.executeTest(TECore.java:811)
	at com.occamlab.te.TECore.execute_test(TECore.java:419)
	at com.occamlab.te.TECore.execute_suite(TECore.java:467)
	at com.occamlab.te.TECore.execute(TECore.java:296)
	at com.occamlab.te.TECore.run(TECore.java:2403)
	at java.lang.Thread.run(Thread.java:748)
Caused by: net.sf.saxon.trans.XPathException: Error in call to extension function {public java.lang.Object com.occamlab.te.TECore.callFunction(net.sf.saxon.expr.XPathContext,java.lang.String,java.lang.String,net.sf.saxon.om.NodeInfo) throws java.lang.Exception}: Exception in extension function net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.RuntimeException: Failed to parse resource located at http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:151)
	at net.sf.saxon.expr.Expression.evaluateItem(Expression.java:352)
	at net.sf.saxon.expr.Expression.process(Expression.java:461)
	at net.sf.saxon.instruct.DocumentInstr.evaluateItem(DocumentInstr.java:282)
	at net.sf.saxon.expr.ExpressionTool.evaluate(ExpressionTool.java:295)
	at net.sf.saxon.expr.LetExpression.eval(LetExpression.java:341)
	at net.sf.saxon.expr.LetExpression.processLeavingTail(LetExpression.java:542)
	at net.sf.saxon.instruct.Block.processLeavingTail(Block.java:409)
	at net.sf.saxon.expr.LetExpression.processLeavingTail(LetExpression.java:551)
	at net.sf.saxon.instruct.Block.processLeavingTail(Block.java:409)
	at net.sf.saxon.instruct.Instruction.process(Instruction.java:94)
	at net.sf.saxon.expr.LetExpression.process(LetExpression.java:380)
	at net.sf.saxon.instruct.ForEach.processLeavingTail(ForEach.java:300)
	at net.sf.saxon.instruct.Template.applyLeavingTail(Template.java:175)
	at net.sf.saxon.instruct.ApplyTemplates.applyTemplates(ApplyTemplates.java:343)
	at net.sf.saxon.Controller.transformDocument(Controller.java:1736)
	at net.sf.saxon.Controller.transform(Controller.java:1560)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:190)
	... 7 more
Caused by: net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.RuntimeException: Failed to parse resource located at http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:192)
	at com.occamlab.te.TECore.executeTemplate(TECore.java:600)
	at com.occamlab.te.TECore.executeXSLFunction(TECore.java:1141)
	at com.occamlab.te.TECore.callFunction(TECore.java:1173)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.sf.saxon.functions.ExtensionFunctionCall.invokeMethod(ExtensionFunctionCall.java:533)
	at net.sf.saxon.functions.ExtensionFunctionCall.call(ExtensionFunctionCall.java:256)
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:147)
	... 24 more
Caused by: net.sf.saxon.trans.XPathException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.RuntimeException: Failed to parse resource located at http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:151)
	at net.sf.saxon.expr.Expression.evaluateItem(Expression.java:352)
	at net.sf.saxon.instruct.SimpleContentConstructor.evaluateItem(SimpleContentConstructor.java:261)
	at net.sf.saxon.instruct.ValueOf.evaluateItem(ValueOf.java:251)
	at net.sf.saxon.instruct.SimpleNodeConstructor.iterate(SimpleNodeConstructor.java:219)
	at net.sf.saxon.instruct.DocumentInstr.evaluateItem(DocumentInstr.java:245)
	at net.sf.saxon.expr.ExpressionTool.evaluate(ExpressionTool.java:295)
	at net.sf.saxon.expr.LetExpression.eval(LetExpression.java:341)
	at net.sf.saxon.expr.LetExpression.process(LetExpression.java:372)
	at net.sf.saxon.instruct.ForEach.processLeavingTail(ForEach.java:300)
	at net.sf.saxon.instruct.Block.processLeavingTail(Block.java:409)
	at net.sf.saxon.instruct.Template.applyLeavingTail(Template.java:175)
	at net.sf.saxon.instruct.ApplyTemplates.applyTemplates(ApplyTemplates.java:343)
	at net.sf.saxon.Controller.transformDocument(Controller.java:1736)
	at net.sf.saxon.Controller.transform(Controller.java:1560)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:190)
	... 34 more
Caused by: java.lang.RuntimeException: Failed to parse resource located at http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities
	at de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.readCapabilities(InteractiveTestUtils.java:105)
	at de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(InteractiveTestUtils.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.sf.saxon.functions.ExtensionFunctionCall.invokeMethod(ExtensionFunctionCall.java:533)
	at net.sf.saxon.functions.ExtensionFunctionCall.call(ExtensionFunctionCall.java:256)
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:147)
	... 49 more
Caused by: java.io.IOException: Server returned HTTP response code: 503 for URL: http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1894)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(Unknown Source)
	at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:177)
	at de.latlon.ets.core.util.URIUtils.resolveURIAsDocument(URIUtils.java:59)
	at de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.readCapabilities(InteractiveTestUtils.java:99)
	... 57 more
```

According to the exception the capability URL is not able to access the resource. I have also tried to get the resource via CURL command but it returning the login page in response. I think the resource URL is not accessible.

Here command: `curl -X GET "http://mdms1-devel1.compusult.net/wes/GeopackageWMS/259?request=GetCapabilities"`

Response screenshot returned via  curl: 
![web enterprise suite](https://user-images.githubusercontent.com/14732650/39307008-4b01e322-497f-11e8-8ef3-c9eda06be9f1.png)

@dstenger 


I have Investigated this issue and tried to reproduce using webapp but I am getting the following error related to getFeatureInfo request:

```
2018/05/15-02:49:20 ERROR
Test Name : tns:Main type Mandatory
net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public java.lang.Object com.occamlab.te.TECore.callFunction(net.sf.saxon.expr.XPathContext,java.lang.String,java.lang.String,net.sf.saxon.om.NodeInfo) throws java.lang.Exception}: Exception in extension function net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.AssertionError: Could not find suitable layer for GetMap requests. expected object to not be null
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:192)
	at com.occamlab.te.TECore.executeTemplate(TECore.java:600)
	at com.occamlab.te.TECore.executeTest(TECore.java:811)
	at com.occamlab.te.TECore.execute_test(TECore.java:419)
	at com.occamlab.te.TECore.execute_suite(TECore.java:467)
	at com.occamlab.te.TECore.execute(TECore.java:296)
	at com.occamlab.te.TECore.run(TECore.java:2403)
	at java.lang.Thread.run(Thread.java:748)
Caused by: net.sf.saxon.trans.XPathException: Error in call to extension function {public java.lang.Object com.occamlab.te.TECore.callFunction(net.sf.saxon.expr.XPathContext,java.lang.String,java.lang.String,net.sf.saxon.om.NodeInfo) throws java.lang.Exception}: Exception in extension function net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.AssertionError: Could not find suitable layer for GetMap requests. expected object to not be null
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:151)
	at net.sf.saxon.expr.Expression.evaluateItem(Expression.java:352)
	at net.sf.saxon.expr.Expression.process(Expression.java:461)
	at net.sf.saxon.instruct.DocumentInstr.evaluateItem(DocumentInstr.java:282)
	at net.sf.saxon.expr.ExpressionTool.evaluate(ExpressionTool.java:295)
	at net.sf.saxon.expr.LetExpression.eval(LetExpression.java:341)
	at net.sf.saxon.expr.LetExpression.processLeavingTail(LetExpression.java:542)
	at net.sf.saxon.instruct.Block.processLeavingTail(Block.java:409)
	at net.sf.saxon.expr.LetExpression.processLeavingTail(LetExpression.java:551)
	at net.sf.saxon.instruct.Block.processLeavingTail(Block.java:409)
	at net.sf.saxon.instruct.Instruction.process(Instruction.java:94)
	at net.sf.saxon.expr.LetExpression.process(LetExpression.java:380)
	at net.sf.saxon.instruct.ForEach.processLeavingTail(ForEach.java:300)
	at net.sf.saxon.instruct.Template.applyLeavingTail(Template.java:175)
	at net.sf.saxon.instruct.ApplyTemplates.applyTemplates(ApplyTemplates.java:343)
	at net.sf.saxon.Controller.transformDocument(Controller.java:1736)
	at net.sf.saxon.Controller.transform(Controller.java:1560)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:190)
	... 7 more
Caused by: net.sf.saxon.s9api.SaxonApiException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.AssertionError: Could not find suitable layer for GetMap requests. expected object to not be null
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:192)
	at com.occamlab.te.TECore.executeTemplate(TECore.java:600)
	at com.occamlab.te.TECore.executeXSLFunction(TECore.java:1141)
	at com.occamlab.te.TECore.callFunction(TECore.java:1173)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.sf.saxon.functions.ExtensionFunctionCall.invokeMethod(ExtensionFunctionCall.java:533)
	at net.sf.saxon.functions.ExtensionFunctionCall.call(ExtensionFunctionCall.java:256)
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:147)
	... 24 more
Caused by: net.sf.saxon.trans.XPathException: Error in call to extension function {public static java.lang.String de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(java.lang.String)}: Exception in extension function java.lang.AssertionError: Could not find suitable layer for GetMap requests. expected object to not be null
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:151)
	at net.sf.saxon.expr.Expression.evaluateItem(Expression.java:352)
	at net.sf.saxon.instruct.SimpleContentConstructor.evaluateItem(SimpleContentConstructor.java:261)
	at net.sf.saxon.instruct.ValueOf.evaluateItem(ValueOf.java:251)
	at net.sf.saxon.instruct.SimpleNodeConstructor.iterate(SimpleNodeConstructor.java:219)
	at net.sf.saxon.instruct.DocumentInstr.evaluateItem(DocumentInstr.java:245)
	at net.sf.saxon.expr.ExpressionTool.evaluate(ExpressionTool.java:295)
	at net.sf.saxon.expr.LetExpression.eval(LetExpression.java:341)
	at net.sf.saxon.expr.LetExpression.process(LetExpression.java:372)
	at net.sf.saxon.instruct.ForEach.processLeavingTail(ForEach.java:300)
	at net.sf.saxon.instruct.Block.processLeavingTail(Block.java:409)
	at net.sf.saxon.instruct.Template.applyLeavingTail(Template.java:175)
	at net.sf.saxon.instruct.ApplyTemplates.applyTemplates(ApplyTemplates.java:343)
	at net.sf.saxon.Controller.transformDocument(Controller.java:1736)
	at net.sf.saxon.Controller.transform(Controller.java:1560)
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:190)
	... 34 more
Caused by: java.lang.AssertionError: Could not find suitable layer for GetMap requests. expected object to not be null
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.assertNotNull(Assert.java:423)
	at de.latlon.ets.wms13.core.util.request.WmsRequestBuilder.buildGetFeatureInfoRequestWithFormat(WmsRequestBuilder.java:128)
	at de.latlon.ets.wms13.core.util.request.WmsRequestBuilder.buildGetFeatureInfoRequest(WmsRequestBuilder.java:83)
	at de.latlon.ets.wms13.core.util.request.WmsRequestBuilder.buildGetFeatureInfoRequest(WmsRequestBuilder.java:64)
	at de.latlon.ets.wms13.core.util.interactive.InteractiveTestUtils.retrieveGetFeatureInfoRequest(InteractiveTestUtils.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.sf.saxon.functions.ExtensionFunctionCall.invokeMethod(ExtensionFunctionCall.java:533)
	at net.sf.saxon.functions.ExtensionFunctionCall.call(ExtensionFunctionCall.java:256)
	at net.sf.saxon.functions.ExtensionFunctionCall.iterate(ExtensionFunctionCall.java:147)
	... 49 more


```


The `<Layer>` are not found in Capability document so it's failing with the above error.

@chuck1987 

Can you please provide more details to reproduce the original issue that you reported?

--- a/src/main/java/de/latlon/ets/wms13/core/dgiwg/testsuite/QueryableWMS.java
+++ b/src/main/java/de/latlon/ets/wms13/core/dgiwg/testsuite/QueryableWMS.java
@@ -140,7 +140,7 @@ public class QueryableWMS {
      *             If an I/O error occurs while trying to read or write the test
      *             run results.
      */
-    @Test(description = "See DGIWG-112: Table 1, Requirement 1")
+    @Test(enabled=false, description = "See DGIWG-112: Table 1, Requirement 1")
     public void invokeBaseTestSuite() throws IOException {
         LOGR.log(logLevel, "Starting base test run at {0}", this.baseTestRunController);
         MultivaluedMap<String, String> args = new MultivaluedMapImpl();

